src/common/res/features/budget-progress-bars/main.js

```diff
@@ -40,46 +53,69 @@
       }
 
       function addGoalProgress(subCategoryName, target) {
-        if ( "Uncategorized Transactions" != subCategoryName ) {
+        if ( 'Uncategorized Transactions' != subCategoryName ) {
           calculation = getCalculation(subCategoryName);
 
-          var status = 0;
-          var hasGoal = false;
-          if (calculation.goalExpectedCompletion > 0 || calculation.goalOverallLeft) {
-            // Target total goal
-            hasGoal = true;
-            status = calculation.balance / (calculation.balance + calculation.goalOverallLeft);
-          }
-          else if (calculation.goalTarget > 0) {
-            // Taget by date
-            // or Montly goal
-            hasGoal = true;
-            status = 1 - calculation.goalUnderFunded / calculation.goalTarget;
-          }
-          else if (calculation.upcomingTransactions < 0) {
-            // Upcoming transactions "goal"
-            hasGoal = true;
-            status = - calculation.balance / calculation.upcomingTransactions;
-          }
+					var status = 0;
+					var tstatus = 0;
+					var hasGoal = false;
 
-          if (hasGoal) {
-            status = status > 1 ? 1 : status;
-            status = status < 0 ? 0 : status;
-            var percent = Math.round(parseFloat(status)*100);
-            target.style.background = "linear-gradient(to right, rgba(22, 163, 54, 0.3) " + percent + "%, white " + percent+ "%)";
-          }
-          else {
-            target.removeAttribute("style");
-          }
-        }
-      }
+					switch (calculation.goalType) {
+						case 'TB' :
+							hasGoal = true;
+
+							if ( calculation.balance >= calculation.targetBalance ) {
+								status = 100;
+							} else {
+								status = calculation.balance / (calculation.balance + calculation.goalOverallLeft);
+							}
+
+							break;
+						case 'TBD' :
```

Comment:
Could this case be combined with the code above? So:

```javascript
case 'TB' :
case 'TBD' :
	hasGoal = true;
  
	if ( calculation.balance >= calculation.targetBalance ) {
		status = 100;
	} else {
		status = calculation.balance / (calculation.balance + calculation.goalOverallLeft);
	}
  
	break;
// Other cases
```

Looks like the same code to me.